lib/isorender/dom-shims.js

```diff
@@ -34,25 +34,51 @@ global.requestAnimationFrame = global.requestAnimationFrame || requestAnimationF
     const child = origFunc.apply(this, arguments);
     requestAnimationFrame(() => {
       if (!child.initialized && child.connectedCallback) {
+        child.isConnected = true;
         child.connectedCallback();
       }
     });
   };
 });
 
+// patch removeChild to call disconnectedCallback
+const origRemoveChild = Element.prototype.removeChild;
+Element.prototype.removeChild = function(child) {
+  origRemoveChild.call(this, child);
+  const childNode = child;
+  if (childNode && childNode.disconnectedCallback && childNode.initialized) {
+    childNode.isConnected = false;
+    childNode.disconnectedCallback();
+  }
+  return child;
+};
+
+Node.DOCUMENT_FRAGMENT_NODE = 11
+
 // html-element only provides Element (with a lot of the HTMLElement API baked in).
 // Use HTMLElement as our Web Components-ready extension.
 class HTMLElement extends Element {
   setAttribute(name, value) {
     const oldValue = this.getAttribute(name);
     super.setAttribute(...arguments);
-    if (this.attributeChangedCallback && this.__attrIsObserved(name)) {
-      this.attributeChangedCallback(name, oldValue, value);
-    }
+    this.__onAttrChanged(name, oldValue, value);
   }
 
-  hasAttribute(name) {
-    return !!this.attributes.find((attr) => attr.name === name);
+  removeAttribute(name) {
+    const oldValue = this.getAttribute(name);
+    super.removeAttribute(...arguments);
+    this.__onAttrChanged(name, oldValue, null);
+  }
+
+  hasAttribute (attr) {
```

Comment:
```suggestion
  hasAttribute(attr) {
```

Reply:
moved this onto `Element.prototype` for a more general shim after some yarn-link testing (with space before parens)